polynote-kernel/src/main/scala/polynote/kernel/remote/transport.scala

```diff
@@ -198,10 +200,23 @@ class SocketTransport(
   forceServerAddress: Option[String] = None
 ) extends Transport[InetSocketAddress] {
 
-  private def openServerChannel: RIO[Blocking, ServerSocketChannel] = effectBlocking {
-    ServerSocketChannel.open().bind(
-      new InetSocketAddress(
-        forceServerAddress.getOrElse(java.net.InetAddress.getLocalHost.getHostAddress), 0))
+  private[this] def socketIsFree(host: String, port: Int): Boolean =
+    try {
+      new Socket(host, port).close()
+      false
+    } catch {
+      case _: ConnectException =>
+        true
+    }
+
+  private def openServerChannel(
+                                 config: PolynoteConfig
```

Comment:
Here you can just make this return `RIO[Blocking with Config]` and do `Config.access` inside, rather than passing the config as a parameter.

Reply:
Ok, I gave it a try based on other Config.access patterns I found in the project. Out of curiosity, what is the difference?